lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1642,9 +1642,101 @@ sub verify_alignment_data {
 
 # This will generally only return something until the first merge has deleted the original per-lane bam file
 sub alignment_bam_file_paths {
-    return glob(File::Spec->join(shift->output_dir, "*.bam"));
+    my $self = shift;
+    return glob(File::Spec->join($self->output_dir, "*.bam"));
+}
+
+
+sub get_bam_file {
+    my $self = shift;
+
+    if ($self->get_merged_alignment_results) {
+        return $self->revivified_alignment_bam_file_path;
+    }
+    else {
+        my $lock = $self->get_bam_lock;
+        if ($self->get_merged_alignment_results) {
+            $lock->unlock;
+            return $self->revivified_alignment_bam_file_path;
+        }
+        else {
+            my @bams = $self->alignment_bam_file_paths;
+            unless (@bams) {
+                $lock->unlock;    
+                if ($self->get_merged_alignment_results) {
```

Comment:
Is it possible for this to become true when it was false on line right after acquiring the lock?